dataprofiler/labelers/character_level_cnn_model.py

```diff
@@ -198,17 +198,23 @@ def set_label_mapping(self, label_mapping):
         :type label_mapping: dict
         :return: None
         """
+        if not isinstance(label_mapping, (list, dict)):
+            raise TypeError("Labels must either be an encoding dict which maps "
+                            "labels to index encodings or a list.")
+
         label_mapping = copy.deepcopy(label_mapping)
-        if not isinstance(label_mapping, dict):
-            raise TypeError("`label_mapping` must be a dict which maps labels "
-                            "to index encodings.")
-        if 'PAD' not in label_mapping and 0 not in label_mapping.values():
-            label_mapping.update({'PAD': 0})
-        if label_mapping.get('PAD', None) != 0:
+        if 'PAD' not in label_mapping:
+            if isinstance(label_mapping, list):  # if list missing PAD
+                label_mapping = ['PAD'] + label_mapping
+            elif 0 not in label_mapping.values():  # if dict missing PAD and 0
+                label_mapping.update({'PAD': 0})
```

Comment:
notice below that it will error out if PAD is not 0
https://github.com/capitalone/DataProfiler/pull/48/files/dcd7a73cc57075ea2866f74146e58a7b99dde343#diff-8ae9227e36e999b076bef5c67ba99f6cf22ecb8c02734691cbf045fb931dfa26R212